### PR TITLE
Catch error from pynvml.nvmlDeviceGetTotalEnergyConsumption (Issue #586)

### PR DESCRIPTION
**Pull request by joshuaj1397**
Added error handling for GPU energy measurements.

### CODE REVIEW

File: codecarbon/core/gpu.py [hunk 1]
```diff
@@ -92,7 +95,10 @@ def _get_total_energy_consumption(self):
         """Returns total energy consumption for this GPU in millijoules (mJ) since the driver was last reloaded
         https://docs.nvidia.com/deploy/nvml-api/group__nvmlDeviceQueries.html#group__nvmlDeviceQueries_1g732ab899b5bd18ac4bfb93c02de4900a
         """
-        return pynvml.nvmlDeviceGetTotalEnergyConsumption(self.handle)
+        try:
+            return pynvml.nvmlDeviceGetTotalEnergyConsumption(self.handle)
+        except pynvml.NVMLError:
+            return None
```
Comment (SaboniAmine) [inline codecarbon/core/gpu.py:101]: Lets add a log, before returning a None value ?

**Review by SaboniAmine — APPROVED** [review]
Thanks for your contribution!

**Comment by SaboniAmine** [comment]
Could you please apply the `black` linter from your fork ? This would solve the pre-commit complaint.

**Comment by benoit-cty** [comment]
I've done the linting, thanks for your contribution.

**Pull request merged**